packages/workflow/src/internals.ts

```diff
@@ -534,14 +542,96 @@ export class Activator implements ActivationHandler {
     );
   }
 
+  public doUpdate(activation: coresdk.workflow_activation.IDoUpdate): void {
+    const { id, name, headers, runValidator } = activation;
+    if (!id) {
+      throw new TypeError('Missing activation update id');
+    }
+    if (!name) {
+      throw new TypeError('Missing activation update name');
+    }
+    if (!this.updateHandlers.has(name)) {
+      // TODO (dan): Signal is able to handle this situation more gracefully by
+      // using this.bufferedSignals. Should something analogous exist for
+      // Update?
+      this.rejectUpdate(id, new ApplicationFailure(`Update has no handler: ${name}`));
```

Comment:
This is a good solution you have here. Many SDKs list all known updates in the error message.

Reply:
I suppose it doesn't make a real difference in this case, but for consistency, please make this `ApplicationFailure.nonRetryable(\`Update has no handler: ${name}\`)` instead. In most places, resulting behavior of `ApplicationFailure`s vary depending on either the failure is retryable or not-retryable, so we're generally avoiding calling `ApplicationFailure`'s constructor directly as it makes the behavior non-obvious for the reader.